### PR TITLE
cmake: add switch to opt out examples from package

### PR DESCRIPTION
**Pull request by nurmukhametov**
Switch ISPC_PACKAGE_EXAMPLES allows to enable/disable packaging examples into ISPC archive. Default is ON.
Add custom target `package-examples` to package them into separate archive from ISPC itself.

### CODE REVIEW

**Review by aneshlya — COMMENTED** [review]
Could you add a CI job which would run `package-examples` target?

File: CMakeLists.txt [hunk 1]
```diff
@@ -763,10 +764,12 @@ endif()
...
 if (ISPC_PREPARE_PACKAGE)
-    if (XE_ENABLED)
-        install (DIRECTORY "${PROJECT_SOURCE_DIR}/examples/" DESTINATION examples)
-    else()
-        install (DIRECTORY "${PROJECT_SOURCE_DIR}/examples/" DESTINATION examples PATTERN "xpu" EXCLUDE)
+    if (ISPC_PACKAGE_EXAMPLES)
+        if (XE_ENABLED)
+            install (DIRECTORY "${PROJECT_SOURCE_DIR}/examples/" DESTINATION examples)
+        else()
+            install (DIRECTORY "${PROJECT_SOURCE_DIR}/examples/" DESTINATION examples PATTERN "xpu" EXCLUDE)
```
Comment (e3m3) [inline CMakeLists.txt:771]: One option to remove redundancy is to set the value of the pattern in this branch and then call `install` once afterwards (with empty pattern for Xe).

File: CMakeLists.txt [hunk 2]
```diff
@@ -878,4 +885,20 @@ if (ISPC_PREPARE_PACKAGE)
     set(CPACK_STRIP_FILES TRUE)
 
     include(CPack)
+
+    # Custom target to pack only examples
+    if (WIN32)
+        add_custom_target(package-examples
+            COMMENT "Package examples directory to ${EXAMPLES_PACKAGE_FILE_NAME}.zip"
+            COMMAND ${CMAKE_COMMAND} -E tar cf "${CMAKE_BINARY_DIR}/${EXAMPLES_PACKAGE_FILE_NAME}.zip" --format=zip examples
+            WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
```
Comment (e3m3) [inline CMakeLists.txt:894]: Didn't know tar handled zips! 👍

**Review by e3m3 — APPROVED** [review]
LGTM

File: CMakeLists.txt (hunk 2 above)
Comment (nurmukhametov) [reply CMakeLists.txt:894]: This is cmake tool: https://cmake.org/cmake/help/latest/manual/cmake.1.html#cmdoption-cmake-E-arg-tar

**Comment by nurmukhametov** [comment]
> Could you add a CI job which would run `package-examples` target?

I have added two jobs: [win-package-examples](https://github.com/ispc/ispc/actions/runs/6262182690/job/17003793345?pr=2658) and [linux-package-examples](https://github.com/ispc/ispc/actions/runs/6262182690/job/17003790854?pr=2658)

**Pull request merged**